### PR TITLE
DAT-2505 Tag Source mode edits (with RTE disabled) as a separate tag

### PR DESCRIPTION
**Pull request by sebastianmarzjan**

### CODE REVIEW

File: extensions/wikia/EditTagging/EditTaggingHooks.class.php [hunk 1]
```diff
@@ -64,26 +66,13 @@ public static function onArticleSaveComplete( &$article, &$user, $text, $summary
...
-			case self::RTE_WYSIWYG_MODE:
-				self::AddRevisionTag( $revision_id, self::RTE_WYSIWYG_MODE_TAG );
-				break;
-			default:
-				break;
-		}
+		self::tagRevisionIfCategoryEdit( $revision_id, $request );
+		self::tagRevisionIfSourceEdit( $revision_id, $request );
+		self::tagRevisionIfRTESourceEdit( $revision_id, $rte_mode );
+		self::tagRevisionIfRTEWysiwygEdit( $revision_id, $rte_mode );
```
Comment (idradm) [inline extensions/wikia/EditTagging/EditTaggingHooks.class.php]: Do we need to go through all four methods each time? Or we just need one?

Comment (SebastianMarzjan) [inline extensions/wikia/EditTagging/EditTaggingHooks.class.php]: We don't. 
I've split this logic to separate methods for readability's sake.

Comment (SebastianMarzjan) [inline extensions/wikia/EditTagging/EditTaggingHooks.class.php]: Ok, moving forward per offline discussion :)

**Comment by idradm** [comment]
LGTM, what about failed tests?

**Comment by SebastianMarzjan** [comment]
Looks like those were temporary problems with branch phpunit test execution. Just ran php-all on this branch with no failures

**Comment by SebastianMarzjan** [comment]
Tests passed, merging

**Pull request merged**